lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/IMRUDriver.cs

```diff
@@ -161,6 +169,12 @@ internal sealed class IMRUDriver<TMapInput, TMapOutput, TResult, TPartitionType>
             _serviceAndContextConfigurationProvider =
                 new ServiceAndContextConfigurationProvider<TMapInput, TMapOutput, TPartitionType>(dataSet, configurationManager);
 
+            if (lifecycleManager != null)
+            {
+                var handle = lifecycleManager.Subscribe(this as IObserver<IJobCancelled>);
+                _disposableResources.Add(handle);
```

Comment:
Yes, when the object is out of scope, the destructor will be called.  I just feel it is safer to call the lifecycleManager.Dispose explicitly in IMRUDriver.DisposeResources() as timer is in a separate thread.  As we throw exception in the cancellation case, I am not sure if the system will call all the destructors properly.